### PR TITLE
Annie/camel case fixes

### PR DESCRIPTION
**Pull request by AnnieLR**
Added notes re camelCase vs capital (import named classes / import default value components)
Plus misc fixes to headings, note style, links.
Please check through my changes. Wasn't sure about all the CSS output since I didn't change the namespace names and some stayed capital.

### CODE REVIEW

File: site/docs/guides/component-variants.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: Component Variants
 layout: docs
 ---
 
-When building a [Stylable component library](./stylable-component-library.md) or a [Stylable application](./stylable-application.md), it is useful to define several semantic "flavors" of some components, for example a `Button` component that represents `cancel` or a `Gallery` component for the `main` gallery instance. 
+When building a [Stylable component library](./stylable-component-library.md) or a [Stylable application](./stylable-application.md), it is useful to define several semantic "flavors" of some components, for example a `button` component that represents `cancel` or a `gallery` component for the `main` gallery instance.
```
Comment (arnonkehat) [inline site/docs/guides/component-variants.md]: Here `button` and `gallery` are components. They need to be capitalized.

File: site/docs/guides/components-basics.md [hunk 2]
```diff
@@ -8,17 +8,17 @@ This guide walks you through the basics of how to style and work with components
 
 You use **Stylable** with a component file (for example using React), along with a **Stylable** CSS file that has the extention `.st.css`.
 
-> **Note**:
+> **Note**:  
 > This guide shows the JSX side of our [stylable-integration](https://github.com/wixplosives/stylable-integration) with React. 
 
-**Stylable** styles are similar to a type-system. Once you have declared that a CSS class is of the type `Button` for example, **Stylable** knows its internal structure and can match its internal parts and states.
+**Stylable** styles are similar to a type-system. Once you have declared that a CSS class is of the type `button` for example, **Stylable** knows its internal structure and can match its internal parts and states.
```
Comment (arnonkehat) [inline site/docs/guides/components-basics.md]: Same here. capitalized.

File: site/docs/guides/components-basics.md [hunk 3]
```diff
@@ -8,17 +8,17 @@ This guide walks you through the basics of how to style and work with components
...
 
 Whether creating your own components or using components you imported from a 3rd party, you want to be able to access and style the internal parts of every component in the scope of your page or application. 
 
 
-## 1 Style a Component 
+## 1 Style a component 
 
-Let's say you have a `Button` component with a render function per this example. You can style its JSX using the `className` property.
+Let's say you have a `button` component with a render function per this example. You can style its JSX using the `className` property.
```
Comment (arnonkehat) [inline site/docs/guides/components-basics.md]: capitalized.

File: site/docs/guides/components-basics.md [hunk 4]
```diff
@@ -86,7 +86,7 @@ render(){
 }
 ```
 
-Let's also import `Button`'s stylesheet into the `Panel` stylesheet. You can then target the internal parts of the component that you imported:
+Let's also import `Button`'s stylesheet into the `panel` stylesheet. You can then target the internal parts of the component that you imported:
```
Comment (arnonkehat) [inline site/docs/guides/components-basics.md]: capitalized.

File: site/docs/guides/stylable-application.md [hunk 5]
```diff
@@ -29,10 +29,10 @@ An application would define CSS with the final style definitions:
 }
 :import {
     -st-from: './button/button.st.css';
-    -st-default: Button;
+    -st-default: button;
```
Comment (arnonkehat) [inline site/docs/guides/stylable-application.md]: capitalize

File: site/docs/guides/stylable-application.md [hunk 6]
```diff
@@ -29,10 +29,10 @@ An application would define CSS with the final style definitions:
...
 }
 .cancelButton {
-    -st-extends: Button;
+    -st-extends: button;
```
Comment (arnonkehat) [inline site/docs/guides/stylable-application.md]: capitalize

File: site/docs/guides/component-variants.md (hunk 1 above)
Comment (AnnieLR) [reply site/docs/guides/component-variants.md]: Fixed

File: site/docs/guides/components-basics.md (hunk 2 above)
Comment (AnnieLR) [reply site/docs/guides/components-basics.md]: Here it says that the "class" is of type button. Shouldn't any "class" be camelCase?

File: site/docs/guides/components-basics.md (hunk 3 above)
Comment (AnnieLR) [reply site/docs/guides/components-basics.md]: Fixed 
(Had debated this one cuz the render function has `<button> ..... </button> with only pseudo-elements between the render & it doesn't say "Button" like other jsx example).

File: site/docs/guides/components-basics.md (hunk 4 above)
Comment (AnnieLR) [reply site/docs/guides/components-basics.md]: Fixed

File: site/docs/guides/stylable-application.md (hunk 5 above)
Comment (AnnieLR) [reply site/docs/guides/stylable-application.md]: Fixed 
(this one I just missed ;-))

File: site/docs/guides/stylable-application.md (hunk 6 above)
Comment (AnnieLR) [reply site/docs/guides/stylable-application.md]: Fixed

File: site/docs/guides/components-basics.md (hunk 2 above)
Comment (arnonkehat) [reply site/docs/guides/components-basics.md]: still a component.

**Pull request merged**